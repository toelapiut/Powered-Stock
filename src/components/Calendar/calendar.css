.calendar {
    border-radius: 5px;
    padding: 30px;
}
.calendar-month {
    width: 300px;
    user-select: none;
}
.calendar-month_header {
    display: flex;
    justify-content: space-between;
    margin-bottom: 30px;
}
.calendar-week,
.calendar-days_of_week {
    display: flex;
    justify-content: space-between;
}
.calendar-week-day,
.calendar-days_of_week_day {
    display: inline-block;
    width: 40px;
    text-align: center;
    line-height: 30px;
    opacity: 0.5;
}
.calendar-week-day {
    margin: 0;
    padding: 0;
    background: none;
    border: none;
}
.calendar-week-day.is-selectable {
    cursor: pointer;
}
.calendar-week-day.is-disabled {
    background: #f5f5f5;
}
.calendar-week-day.is-current_month {
    opacity: 1;
}
.calendar-week-day.is-selected {
    background: #008000;
}
.calendar-week-day.is-selected.is-start_selection {
    background: #f00;
}
.calendar-week-day.is-selected.is-end_selection {
    background: #00f;
}
.calendar-header_button.is-prev:before {
    content: '<';
}
.calendar-header_button.is-next:before {
    content: '>';
}
