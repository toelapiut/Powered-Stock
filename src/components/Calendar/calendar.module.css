.container{
    @apply bg-koamaru backdrop-filter backdrop-blur-2xl w-full z-10 flex items-center justify-center
}