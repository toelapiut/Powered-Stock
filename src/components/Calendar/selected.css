.date_picker {
    @apply p-16 border border-navy bg-navy;
    padding: 30px;
    border-radius: 5px;
    /*border: 1px solid #e4e7ea;*/
    color: #303030;
    margin-top: 10px;
    display: flex;
    flex-direction: column;
    flex-basis: 310px;
    max-width: 310px;
    width:310px;
}
.date_picker-month_header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 20px;
}
.date_picker-month_header_wrapper {
    text-align: center;
    margin: 0 auto 25px;
}
.date_picker-month_header_title {
    @apply text-base font-black
}
.date_picker-header_button {
    @apply cursor-pointer rounded-full border border-koamaru;
    transition: border 0.1s ease-in, color 0.1s ease-in;
    position: relative;
    height: 35px;
    width: 35px;
}
.date_picker-header_button:focus {
    outline: 0;
}
.date_picker-header_button.is-disabled {
    cursor: not-allowed;
}
.date_picker-header_button.is-next:before,
.date_picker-header_button.is-prev:before {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    transition: fill 0.1s ease-in;
    color: #ffffff;
}
.date_picker-header_button.is-disabled .date_picker-header_button.is-next,
.date_picker-header_button.is-disabled .date_picker-header_button.is-prev {
    color: #acb3bb;
}
.date_picker-calendar_wrapper.is-shortcut_hover_outrange .date_picker-header_button.is-next,
.date_picker-calendar_wrapper.is-shortcut_hover_outrange .date_picker-header_button.is-prev {
    border-color: #4a90e2;
    background: #e7f5ff;
}
.date_picker-header_button.is-prev:before {
    content: '<';
}
.date_picker-header_button.is-next:after {
    content: '>';
}
.date_picker-days_of_week {
    font-size: 13px;
    display: flex;
    color: #c3c3c3;
    text-align: center;
    margin-bottom: 25px;
}
.date_picker-days_of_week_day {
    flex-basis: 15%;
}
.date_picker-week {
    display: flex;
    margin-bottom: 8px;
}

.date_picker-week:last-of-type {
    margin-bottom: 0;
}

.date_picker-week-day {
    height: 34px;
    width: 34px;
    vertical-align: middle;
    font-size: 15px;
    user-select: none;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-basis: 15%;
    position: relative;
    border-radius: 50%;
    margin: 0;
    padding: 0;
    border: none;
    outline: 0;
}
.date_picker-week-day:first-of-type:hover,
.date_picker-week-day:last-of-type:hover {
    border-radius: 50%;
}
.date_picker-week-day:first-of-type {
    border-radius: 50% 0 0 50%;
}
.date_picker-week-day:last-of-type {
    border-radius: 0 50% 50% 0;
}
.date_picker-week-day.is-prev_month,
.date_picker-week-day.is-next_month {
    @apply text-koamaru
}
.date_picker-week-day.is-today {
    font-weight: 700;
}
.date_picker-week-day.is-selected {
    @apply bg-koamaru font-bold text-white relative;
    z-index: 1;
    border-radius: 0;
}
.date_picker-week-day.is-selected:first-of-type {
    border-radius: 50% 0 0 50%;
}
.date_picker-week-day.is-selected:last-of-type {
    border-radius: 0 50% 50% 0;
}
.date_picker-week-day.is-selected::before {
    @apply bg-white rounded-full absolute text-white;
    width: 30px;
    height: 30px;
    z-index: -1;
}
.date_picker-calendar_wrapper.is-shortcut_hovered .date_picker-week-day.is-selected::before {
    background: #4a90e2;
}
.date_picker-week-day.is-selected:hover::before {
    content: '';
    box-shadow: inset 0 0 1px 0 #3863a0;
    background: #e3effc;
    color: #404040;
}
.date_picker-week-day.is-selected.is-start_selection.date_picker-week-day.is-selected:hover::before,
.date_picker-week-day.is-selected.is-end_selection.date_picker-week-day.is-selected:hover::before {
   @apply bg-white;
}
.date_picker-week-day.is-selected.is-start_selection,
.date_picker-week-day.is-selected.is-end_selection {
    @apply text-navy font-bold;
}
.date_picker-week-day.is-selected.is-start_selection::before,
.date_picker-week-day.is-selected.is-end_selection::before {
    content: '';
}
.date_picker-week-day.is-selected.is-start_selection {
    border-radius: 50% 0 0 50%;
}
.date_picker-week-day.is-selected.is-start_selection:last-of-type {
    border-radius: 50%;
}
.date_picker-week-day.is-selected.is-end_selection {
    border-radius: 0 50% 50% 0;
}
.date_picker-week-day.is-selected.is-end_selection.is-start_selection,
.date_picker-week-day.is-selected.is-end_selection:first-of-type {
    @apply rounded-full;
}

.date_picker-week-day.is-selected.is-end_selection.is-start_selection:focus,
.date_picker-week-day.is-selected.is-end_selection:focus {
    @apply outline-none;
}
.date_picker-week-day.is-selectable {
    cursor: pointer;
    position: relative;
}
/*.date_picker-week-day.is-selectable:before {*/
/*    top: 0;*/
/*    left: 0;*/
/*    bottom: 0;*/
/*    right: 0;*/
/*    display: block;*/
/*    width: 100%;*/
/*    height: 100%;*/
/*    position: absolute;*/
/*}*/
.date_picker-week-day.is-selectable:hover:not(.is-selected) {
    z-index: 1;
    box-shadow: inset 0 0 1px 0 #3863a0;
}
.date_picker-week-day.is-not_selectable {
    color: #073058;
    cursor: not-allowed;
}
.date_picker-notice {
    font-size: 11px;
    font-weight: 700;
    margin-bottom: 15px;
    border-radius: 5px;
    padding: 15px;
    color: #fff;
    background: #ff6c7e;
    max-width: 238px;
}
