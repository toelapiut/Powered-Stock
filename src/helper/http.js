import axios from 'axios';

export const quandl = axios.create({
  baseURL: process.env.process.env.REACT_APP_BASE_URL,
  headers: {
    'Content-Type': 'application/json',
    'Accept': 'application/json',
    'Allow': 'GET, PUT, HEAD, OPTIONS'
  }
});

quandl.interceptors.request.use((config) => {
  config.params = config.params || {};
  config.params['key'] = process.env.REACT_APP_API_KEY;
  return config;
});
