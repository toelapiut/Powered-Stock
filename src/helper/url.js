export const url = {
  markets: 'datatables'
};

export const types = {
  ZACKS_CP: 'ZACKS/CP',
  ZACKS_MT: 'ZACKS/MT'
};