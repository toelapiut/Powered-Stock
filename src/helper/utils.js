import humps from 'humps';
import checkPropTypes from 'check-prop-types';
import {timeFormat} from 'd3-time-format';
import {bisector} from 'd3-array';
import moment from 'moment';

export const transpose = (rows, cols) => {
  /*
   * check if the row and col are empty and throw an error to
   * inform of the necessary requirements
   */
  if (typeof rows === 'undefined' || typeof cols === 'undefined') {
    throw new Error('row and col can not be undefined');
  }

  /*
   * throw error if the array isn't an array or col isn't an array
   */
  if (!Array.isArray(rows)) throw new Error('row must be  array');
  if (!Array.isArray(cols)) throw new Error('col must be  array');

  let dataset = [];

  for (let row of rows) {
    /*
     * check the data type before iterating through the matrix
     */
    if (!Array.isArray(row)) throw new Error('row must be a matrix row: [[a, b]]');
    let obj = {};
    row.forEach((item, index) => {
      obj[cols[index]['name']] = item;
    });
    dataset.push(obj);
  }

  return dataset;
};


export const transposeStock = (rows) => {
  let data = [];
  for (let row of rows) {
    let obj = {};
    const [date, close] = row;
    obj['date'] = date;
    obj['close'] = close;
    data.push(obj);
  }
  return data;
};

export const toCamelCase = (item) => {
  if (typeof item === 'undefined') {
    return item;
  }
  return humps.camelizeKeys(item);
};

export const checkPositivity = (num) => {
  if (typeof num === 'undefined') throw new Error('argument must not be empty');
  if (typeof num !== 'number') throw new Error('argument is not a number');
  let check = Math.sign(Number(num));
  return check !== -1;
};

export const formatDate = timeFormat('%b %d, \'%y');

export const getDate = (d) => new Date(d.date);

export const getStockValue = (d) => d.close;

export const bisectDate = bisector(d => new Date(d.date)).left;

export const getWindowDimensions = () => {
  const {innerWidth: width, innerHeight: height} = window;
  return {
    width,
    height
  };
};

export const dateFormat = (date, format = 'DD MMM, YYYY') => {
  return moment(date).format(format);
};

export const findTestByAttr = (component, attr) => {
  return component.find(`[data-test='${attr}']`);
};

export const checkProps = (component, expectedProps) => {
  checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
};

