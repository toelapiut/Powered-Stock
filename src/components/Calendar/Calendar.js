import React from 'react';
import PropTypes from 'prop-types';
import SimpleReactCalendar from 'simple-react-calendar';
import './calendar.css';
import './selected.css';
import styles from './calendar.module.css';

export const Calendar = ({isOpen, onChangeDates, start, end}) => {

  console.log({isOpen});
  return (
    <div>
      <button className={ styles.container}>
        <SimpleReactCalendar
          mode='range'
          minNumberOfWeeks={6}
          blockClassName='date_picker'
          selected={{start, end}}
          onSelect={(date) => onChangeDates(date)}
        />
      </button>
    </div>
  );
};


Calendar.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  onChangeDates: PropTypes.func.isRequired,
  start: PropTypes.object.isRequired,
  end: PropTypes.object.isRequired,
};

