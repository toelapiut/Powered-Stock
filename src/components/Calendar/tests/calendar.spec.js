import React from 'react';
import Calendar from '../';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {findTestByAttr, checkProps} from '../../../helper/utils';


const setUp = (props = {}) => {
  return shallow(<Calendar {...props}/>);
};

let props = {
  date: {
    start: '01-04-2020',
    end: '01-04-2020'
  },
  name: 'Powered By People Inc',
  ticker: 'PBP',
  start: '01-04-2020',
  end: '01-04-2020',
  isOpen: false,
  onChangeDates: () => null,
  onOpenCalendar: () => null,
};

describe('Component Component', () => {
  describe('calendar renders tests', () => {
    let component;
    beforeAll(() => {
      component = setUp(props);
    });

    test('should render snapshot correctly', () => {
      expect(toJson(component)).toMatchSnapshot();
    });

    it('should render without error', () => {
      const wrapper = findTestByAttr(component, 'calendar-container');
      expect(wrapper.length).toBe(1);
    });
  });

  describe('Check property types', () => {
    it('should not throw a warning', () => {
      const propErr = checkProps(Calendar, props);
      expect(propErr).toBeUndefined();
    });
  });
});