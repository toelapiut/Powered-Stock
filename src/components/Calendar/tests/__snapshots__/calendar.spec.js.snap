// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Component calendar renders tests should render snapshot correctly 1`] = `
<div
  data-test="calendar-container"
>
  <button
    className="container"
  >
    <Calendar
      blockClassName="date_picker"
      daysOfWeek={
        Array [
          "Mon",
          "Tue",
          "Wed",
          "Thu",
          "Fri",
          "Sat",
          "Sun",
        ]
      }
      disableDaysOfWeek={false}
      getDayFormatted={[Function]}
      getISODate={[Function]}
      getNoticeContent={[Function]}
      headerNextTitle="Next month"
      headerPrevTitle="Previous month"
      maxDate="01-04-2020"
      minDate="01-04-2020"
      minNumberOfWeeks={6}
      mode="range"
      onSelect={[Function]}
      renderDay={[Function]}
      renderDayOfWeek={[Function]}
      renderNotice={[Function]}
      selected={
        Object {
          "end": "01-04-2020",
          "start": "01-04-2020",
        }
      }
      today="01-04-2020"
      weekStartsOn={1}
    />
  </button>
</div>
`;
